### PR TITLE
add gradient boosting content

### PR DESCRIPTION
**Pull request by LongfeiH**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [X] review the artifacts from local build
- [X] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [X] Unit test if appliable

# Checklist:

- [X] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have applied tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by reveurmichael** [comment]
If I am not getting things wrong, we should have already the titanic dataset. Maybe we could reuse that?

**Comment by reveurmichael** [comment]
the filename "train.csv" doesn't make any sense. either rename it, either put it in a folder named "gradient-descent"

File: open-machine-learning-jupyter-book/ml-advanced/gradient-boosting/xgboost.md [hunk 1]
```diff
@@ -15,479 +15,128 @@ kernelspec:
...
-- The tree based algorithms have evolved over the years.
-
-- XGBoost belongs to a family of tree-based algorithms.
-
-- Please see the chart below for the evolution of tree-based algorithms over the years.
+XGBoost is an implementation of the **Gradient Boosted Decision Trees** algorithm (scikit-learn has another version of this algorithm, but XGBoost has some technical advantages.)  What is **Gradient Boosted Decision Trees**?  We'll walk through a diagram.
 
 :::{figure-md}
-<img src="https://miro.medium.com/max/925/1*QJZ6W-Pck_W7RlIDwUIN9Q.jpeg" width="90%" class="bg-white mb-1">
+<img src="https://i.imgur.com/e7MIgXk.png" width="90%" class="bg-white mb-1">
```
Comment (reveurmichael) [inline open-machine-learning-jupyter-book/ml-advanced/gradient-boosting/xgboost.md:31]: download the image to local

Comment (reveurmichael) [inline open-machine-learning-jupyter-book/ml-advanced/gradient-boosting/xgboost.md:31]: download the image to local

**Comment by LongfeiH** [comment]
> If I am not getting things wrong, we should have already the titanic dataset. Maybe we could reuse that?

This is a ready-to-train version: doesn't not contains missings and categorical data has been encoded.So the two datasets are different.

File: open-machine-learning-jupyter-book/ml-advanced/gradient-boosting/xgboost.md (hunk 1 above)
Comment (LongfeiH) [reply open-machine-learning-jupyter-book/ml-advanced/gradient-boosting/xgboost.md:31]: done,i download all images

Comment (LongfeiH) [reply open-machine-learning-jupyter-book/ml-advanced/gradient-boosting/xgboost.md:31]: done,i download all images

**Review by reveurmichael — APPROVED** [review]

**Pull request merged**